### ISSUE TITLE
False positive `no-member` when attempting to access `_asdict`, which is a valid `NamedTuple` method, in a subclass of `NamedTuple`

### ISSUE THREAD

**Issue by Pierre-Sassoulas** [PyCQA/pylint#7891]
Possibly related to #2688: the same `no-member` behavior occurs when attempting to access `_asdict`, which is a valid `NamedTuple` method, in a subclass of `NamedTuple`:

```python
from typing import NamedTuple


class Test(NamedTuple):
    a: int
    b: str

    def foo(self):
        return self._asdict()  # E1101: Instance of 'Test' has no '_asdict' member (no-member)
```

```
❯ pylint --version
pylint 2.8.0
astroid 2.5.5
Python 3.8.9 (default, Apr  3 2021, 01:50:09)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

_Originally posted by @whoburg in https://github.com/PyCQA/pylint/issues/2688#issuecomment-826363470_

**Comment by manav-glean** [comment]
@Pierre-Sassoulas any update on this?

**Comment by One-Nose** [comment]
I also encountered this problem. Also occurrs with `_replace`, iteration (though not with `iter()`) and probably other attributes as well.

**Comment by mbyrnepr2** [comment]
Note also the `_fields` attribute also reproduces this.
